### PR TITLE
feat(type): Uses CtTypeAccess for static field read / field write / calls.

### LINKED ISSUE #346: Target type of a static field must be a CtTypeAccess
see test TargetedExpressionTest::testStaticTargets
in https://github.com/monperrus/spoon/tree/target

### LINKED ISSUE #476: implicit this access
(follow up of http://lists.gforge.inria.fr/pipermail/spoon-discuss/2016-January/001843.html)

current behavior: when this is implicit getTarget() returns null.

following the work on CtTypeAccess for static fields and methods, we can add a contract
''getTarget() never returns null'' and "getTarget() returns a CtThisAccess when this is implicit."

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #346
Closes #476

### CODE REVIEW

File: src/test/java/spoon/test/targeted/TargetedExpressionTest.java [hunk 1]
```diff
@@ -92,4 +99,36 @@ public void testNotTargetedExpression() throws Exception {
...
+		assertEquals(expectedType, elements.get(2).getVariable().getDeclaringType());
+
+		// contract for static calls getTarget() == null
+		final CtThisAccess<Foo> exepectedThisAccess = type.getFactory().Core().createThisAccess();
+		exepectedThisAccess.setType(expectedType);
+		final CtTypeAccess<Foo> expectedTypeAccess = type.getFactory().Code().createTypeAccess(expectedType);
+
+		assertEquals(exepectedThisAccess, elements.get(0).getTarget());
+		assertEquals(expectedTypeAccess, elements.get(1).getTarget());
+		assertEquals(expectedTypeAccess, elements.get(2).getTarget());
```
Comment (monperrus) [inline src/test/java/spoon/test/targeted/TargetedExpressionTest.java]: add an assert for the getTarget of the 3 field writes?

Comment (GerardPaligot) [inline src/test/java/spoon/test/targeted/TargetedExpressionTest.java]: Done.

**Comment by GerardPaligot** [comment]
@monperrus PR ok for me.

**Comment by GerardPaligot** [comment]
Adds javadoc in CtTargetedExpression : getTarget returns never null and add some samples.

**Pull request merged**